File: src/main/java/io/github/aphelium/modules/brokers/useCases/listById/ListByIdUseCase.java
package io.github.aphelium.modules.brokers.useCases.listById;

import io.github.aphelium.modules.brokers.entities.Broker;
import io.github.aphelium.modules.brokers.repositories.BrokerRepository;
import io.github.aphelium.modules.brokers.exceptions.BrokerNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.UUID;

@Service
public class ListByIdUseCase {

    final BrokerRepository brokerRepository;

    public ListByIdUseCase(BrokerRepository brokerRepository) {
        this.brokerRepository = brokerRepository;
    }

    public Optional<Broker> execute(UUID id) {

        Broker broker = brokerRepository.findById(id).orElseThrow(() -> new BrokerNotFoundException());

        return Optional.ofNullable(broker);

    }

}

File: src/main/kotlin/io/github/aphelium/modules/brokers/useCases/listById/ListByIdUseCase.kt
package io.github.aphelium.modules.brokers.useCases.listById

import io.github.aphelium.modules.brokers.entities.Broker
import io.github.aphelium.modules.brokers.exceptions.BrokerNotFoundException
import io.github.aphelium.modules.brokers.repositories.BrokerRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class ListByIdUseCase(val brokerRepository: BrokerRepository) {
    fun execute(id: UUID): Broker {
        return brokerRepository.findById(id).orElseThrow { BrokerNotFoundException() }!!
    }
}